python/podio_class_generator.py

```diff
@@ -222,15 +229,42 @@ def _process_datatype(self, name, definition):
     self._fill_templates('Obj', datatype)
     self._fill_templates('Collection', datatype)
     self._fill_templates('CollectionData', datatype)
+    self._fill_templates('MutableStruct', datatype)
+    self._fill_templates('Constructor', datatype)
 
     if 'SIO' in self.io_handlers:
       self._fill_templates('SIOBlock', datatype)
 
+  def _preprocess_for_julia(self, datatype):
+    """Do the preprocessing that is necessary for Julia code generation"""
+    includes_jl = set()
+    for relation in datatype['OneToManyRelations'] + datatype['VectorMembers'] + datatype['OneToOneRelations']:
+      if self._needs_include(relation):
+        if not relation.is_builtin:
+          if relation.full_type == datatype['class'].full_type:
+            includes_jl.add(self._build_include(datatype['class'].bare_type, julia=True, is_struct=True))
+          else:
+            includes_jl.add(self._build_include(relation.bare_type, julia=True, is_struct=True))
+          # if datatype['class'].bare_type != relation.bare_type:
+          #   includes_jl.add(self._build_include(relation.bare_type + 'Collection', julia=True))
+    try:
+      includes_jl.remove(self._build_include(datatype['class'].bare_type, julia=True, is_struct=True))
+    except KeyError:
+      pass
+    datatype['includes_jl']['constructor'].update((includes_jl))
+
+  def _get_julia_params(self, datatype):
+    """Get the parameteric types for MutableStructs"""
+    params = set()
+    for relation in datatype['OneToManyRelations'] + datatype['VectorMembers'] + datatype['OneToOneRelations']:
```

Comment:
```suggestion
    """Get the relations as parameteric types for MutableStructs"""
    params = set()
    for relation in datatype['OneToManyRelations'] + datatype['OneToOneRelations']:
```
Just to make the docstring a bit more clear. 
Also I am not sure we actually discussed this during prototyping, but looking at the code now, I just realize that the `VectorMembers` are not needed here, because they can only be builtin types, or components. This means that there should really be no way to end up with a cyclic dependency.